projects/kit/src/lib/time/processors/zero-padding-postprocessor.ts

```diff
@@ -0,0 +1,46 @@
+import {MaskitoOptions} from '@maskito/core';
+import {MaskitoTimeSegments} from '../types';
+import {parseTimeString, toTimeString} from '../utils';
+
+export function createZeroPaddingPostprocessor(
+    timeSegmentMaxValues: MaskitoTimeSegments<number>,
+): NonNullable<MaskitoOptions['postprocessor']> {
+    return elementState => {
+        const {value, selection} = elementState;
+        const {hours, minutes, seconds, milliseconds} = parseTimeString(value);
+
+        const newTimeString = toTimeString({
+            hours: padWithZeroes(hours, `${timeSegmentMaxValues.hours}`.padStart(2, '0')),
+            minutes: padWithZeroes(
+                minutes,
+                `${timeSegmentMaxValues.minutes}`.padStart(2, '0'),
+            ),
+            seconds: padWithZeroes(
+                seconds,
+                `${timeSegmentMaxValues.seconds}`.padStart(2, '0'),
+            ),
+            milliseconds: padWithZeroes(
+                milliseconds,
+                `${timeSegmentMaxValues.milliseconds}`.padStart(3, '0'),
+            ),
+        });
+
+        const newCharactersAmount = newTimeString.length - value.length;
+        const [from] = selection;
+
+        return {
+            value: newTimeString,
+            selection: [from + newCharactersAmount, from + newCharactersAmount],
+        };
+    };
+}
+
+function padWithZeroes(timeSegment: string, maxSegmentValue: string): string {
```

Comment:
Shouldn't it just be [padStart](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart) somehow?

Reply:
It can be possible if all time segments are the same length (but they can be 2 or 3-character length).

That is why I need recursive function here.